src/odemis/acq/align/__init__.py

```diff
@@ -153,35 +147,42 @@ def _cancelTurnLightAndCheck(f):
             logging.debug("The task already finished")
             return False
         f._task_state = CANCELLED
-        f._done.set()
         logging.debug("Task cancelled.")
     return True
 
 
-def _doTurnLightAndCheck(f, bl, ccd):
-    f._done.clear()
+def _doTurnOnLightAndCheck(f, bl, ccd):
     try:
         # check if the light is already turned on. The light should normally be turned off.
         # In case it's already turned on return and start the procedure of autofocus
         if bl.emissions.value[0] * bl.power.value != 0:
+            logging.debug("The light is already on")
             return
         if f._task_state == CANCELLED:
             raise CancelledError()
 
         # Light turned off, take the avg intensity of the image
-        img_light_off = ccd.data.get()
+        img_light_off = ccd.data.get(asap=False)
         avg_intensity_off = numpy.average(img_light_off)
         # set the avg minimum intensity - a significant change (+50%)
         avg_intensity_min_on = avg_intensity_off * 1.5 + 0.1
         # Turn the light on
         bl.power.value = bl.power.range[1]
         bl.emissions.value = [1] * len(bl.emissions.value)
-        img2 = ccd.data.get()
-        # Check whether 0.5 percent of the total pixels have intensity more than the avg minimum
-        while (numpy.sum(img2 > avg_intensity_min_on))/(img2.shape[0]*img2.shape[1]) < 0.005:
+        while True:
             img2 = ccd.data.get()
-        logging.debug("the light is on")
-        return
+            # number of pixels with higher intensity than the avg minimum
```

Comment:
You need to subtract the original image. Otherwise, even a "dark" image could be considered containing light.
Also, you need to protect in case the subtraction fails (eg, because the CCD resolution has changed), and in this case, you could fallback to a simpler version which just subtracts the `avg_intensity_off`.